tests/org/intellij/erlang/resolve/ErlangFunctionResolutionTest.java

```diff
@@ -32,32 +42,79 @@ protected String getTestDataPath() {
+                        String focusedFilePath, String expectedFilePath, String... otherFilePaths) {
     String[] paths = {focusedFilePath, expectedFilePath};
     PsiFile[] files = myFixture.configureByFiles(ArrayUtil.mergeArrays(paths, otherFilePaths));
     ErlangFile focusedFile = (ErlangFile) files[0];
-    assertNotNull(focusedFile);
     ErlangFile expectedFile = (ErlangFile) files[1];
     assertNotNull(expectedFile);
 
     int offset = myFixture.getEditor().getCaretModel().getOffset();
-    ErlangFunctionCallExpression functionCall = PsiTreeUtil.getParentOfType(focusedFile.findElementAt(offset),
-                                                                            ErlangFunctionCallExpression.class);
-    assertNotNull(functionCall);
-    PsiReference reference = functionCall.getReference();
-    assertNotNull(reference);
-    PsiElement resolvedFunction = reference.resolve();
+    PsiElement focusedElement = PsiTreeUtil.getParentOfType(focusedFile != null ? focusedFile.findElementAt(offset)
+                                                                                : null, psiClass);
+    PsiReference reference = focusedElement != null ? focusedElement.getReference() : null;
+    PsiElement resolvedFunction = reference != null ? reference.resolve() : null;
     assertNotNull(resolvedFunction);
 
-    String name = functionCall.getName();
-    int arity = functionCall.getArgumentList().getExpressionList().size();
-    ErlangFunction expectedFunction = expectedFile.getFunction(name, arity);
+    Pair<String, Integer> nameAndArity = getNameAndArity(focusedElement);
+    ErlangFunction expectedFunction = expectedFile.getFunction(nameAndArity.first, nameAndArity.second);
+    if (expectedFunction == null) {
+      expectedFunction = expectedFile.getFunctionsByName(nameAndArity.first).iterator().next();
+    }
     assertSame(expectedFunction, resolvedFunction);
   }
 
-  public void testNoAutoImport()               { doTest("test.erl", "incl.erl"); }
-  public void testNoAutoImportWithTuple()      { doTest("test.erl", "incl.erl"); }
-  public void testPreferImportFromUserModule() { doTest("test.erl", "incl.erl"); }
-  public void testPreferLocalFunction()        { doTest("test.erl", "test.erl", "incl.erl"); }
-  public void testPreferFirstImport()          { doTest("test.erl", "incl.erl", "another_incl.erl"); }
+  @NotNull
+  private static Pair<String, Integer> getNameAndArity(@Nullable PsiElement element) {
+    ErlangFunctionCallExpression call = ObjectUtils.tryCast(element, ErlangFunctionCallExpression.class);
+    if (call != null) {
+      return Pair.create(call.getName(), call.getArgumentList().getExpressionList().size());
+    }
+    ErlangQAtom atom = ObjectUtils.tryCast(element, ErlangQAtom.class);
+    if (atom != null ) {
+      ErlangFunctionReference ref = ErlangPsiImplUtil.createFunctionReference(atom);
+      return Pair.create(ref.getName(), ref.getArity());
+    }
+    return Pair.create("", -1);
```

Comment:
This is crazy. It's as if you're writing in some dynamically-typed language. Use static types where possible.